sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/BinaryOperationWithIdenticalExpressions.cs

```diff
@@ -40,6 +42,25 @@ public void Test(bool a, bool b)
             int i = 1 << 1;
             i = 1 << 0x1;
             i = 2 << 2; // Compliant
+
+            object.Equals(i, i);
+//                        ^ {{Correct one of the identical expressions on both sides of Object.Equals.}}
+//                           ^ Secondary@-1
+
+            var o = new object();
+
+            o.Equals(o);
+//          ^ {{Correct one of the identical expressions on both sides of Object.Equals.}}
+//                   ^ Secondary@-1
+
+            (new object()).Equals(new object());

```

Comment:
Ask Jesse about this case. Shall we report or not? The expression is the same but not the actual values. Although I agree that this is probably a fishy code.